src/test/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtilTest.java

```diff
@@ -15,96 +13,218 @@
 
+    public static final String ID_1 = "17a74540-c1b6-49e2-a81b-a9dbd2259251";
+    public static final String ID_2 = "2ca270ca-0738-4536-8846-7cea34ff8762";
+    public static final String ID_3 = "3a9c1e2a-9536-4aa2-b63d-7cd874e582e3";
+    public static final String ID_4 = "440d0d83-75e1-466a-bacc-90ce9e612074";
+    public static final String ID_5 = "569df4d6-2e1a-43af-a188-8c725d40f8d5";
+
     @Test
     public void givenUcbIsYesForOneOtherParty_thenSetCaseDataOtherPartyUcb() {
-        SscsCaseData data = SscsCaseData.builder().otherParties(Arrays.asList(buildOtherParty("1", true), buildOtherParty("2", false))).build();
+        SscsCaseData data = SscsCaseData.builder().otherParties(Arrays.asList(
+            buildOtherParty(ID_1, true),
+            buildOtherParty(ID_2, false))).build();
         OtherPartyDataUtil.updateOtherPartyUcb(data);
         assertEquals(YesNo.YES.getValue(), data.getOtherPartyUcb());
     }
 
     @Test
     public void givenUcbIsNoForAllOtherParty_thenSetCaseDataOtherPartyUcb() {
-        SscsCaseData data = SscsCaseData.builder().otherParties(Arrays.asList(buildOtherParty("1", false), buildOtherParty("2", false))).build();
+        SscsCaseData data = SscsCaseData.builder().otherParties(Arrays.asList(
+            buildOtherParty(ID_1, false),
+            buildOtherParty(ID_2, false))).build();
         OtherPartyDataUtil.updateOtherPartyUcb(data);
         assertEquals(YesNo.NO.getValue(), data.getOtherPartyUcb());
     }
 
     @Test
     @Parameters({"UPDATE_OTHER_PARTY_DATA", "DWP_UPLOAD_RESPONSE"})
     public void givenNewOtherPartyAdded_thenAssignAnIdAndNotificationFlag(EventType eventType) {
-        List<CcdValue<OtherParty>> otherParties = Arrays.asList(buildOtherPartyWithAppointeeAndRep(null, null, null));
 
-        OtherPartyDataUtil.assignNewOtherPartyData(otherParties, eventType);
 
-        assertEquals(1, otherParties.size());
-        assertEquals("1", otherParties.get(0).getValue().getId());
-        assertEquals("2", otherParties.get(0).getValue().getAppointee().getId());
-        assertEquals("3", otherParties.get(0).getValue().getRep().getId());
-        assertTrue(YesNo.isYes(otherParties.get(0).getValue().getSendNewOtherPartyNotification()));
+        List<CcdValue<OtherParty>> otherParties = Arrays.asList(
+            buildOtherPartyWithAppointeeAndRep(null, null, null));
+
+        OtherPartyDataUtil.assignNewOtherPartyData(otherParties);
+
+        assertThat(otherParties)
+            .hasSize(1)
+            .extracting(CcdValue::getValue)
+            .extracting(OtherParty::getSendNewOtherPartyNotification)
+            .containsOnly(YES);
+
+        assertThat(otherParties)
+            .extracting(CcdValue::getValue)
+            .flatExtracting(otherParty -> List.of(otherParty.getAppointee(), otherParty.getRep()))
+            .extracting(Entity::getId)
+            .hasSize(2)
+            .doesNotContainNull()
+            .extracting(String::length)
```

Comment:
Probably not critical but the id assertion on the otherparty itself is missing

Reply:
this should be fixed with the test rework